editoast/src/views/timetable/train_schedule.rs

```diff
@@ -817,4 +1199,282 @@ pub mod tests {
+            .core_client(core.into())
+            .build();
+
+        let request = app.post("/train_schedule/track_occupancy").json(
+            &json!({"train_schedule_ids": [train_schedule_valid.id],
+                "operational_point_id": "fake_station_id",
+                "infra_id": small_infra.id
+            }),
+        );
+
+        app.fetch(request).assert_status(StatusCode::NOT_FOUND);
+    }
+
+    #[rstest]
+    async fn test_track_occupancy_pathfinding_failure() {
+        let db_pool = DbConnectionPoolV2::for_tests();
+
+        let rolling_stock =
+            create_fast_rolling_stock(&mut db_pool.get_ok(), "simulation_rolling_stock").await;
+        let timetable = create_timetable(&mut db_pool.get_ok()).await;
+        let train_schedule_base = TrainSchedule {
+            rolling_stock_name: rolling_stock.name.clone(),
+            ..serde_json::from_str(include_str!("../../tests/train_schedules/simple_2.json"))
+                .expect("Unable to parse")
+        };
+
+        let train_schedule: Changeset<models::TrainSchedule> = TrainScheduleForm {
+            timetable_id: Some(timetable.id),
+            train_schedule: train_schedule_base.clone(),
+        }
+        .into();
+        let train_schedule_valid = train_schedule
+            .create(&mut db_pool.get_ok())
+            .await
+            .expect("Failed to create train schedule");
+
+        let mut core = MockingClient::new();
+        core.stub("/pathfinding/blocks")
+            .method(reqwest::Method::POST)
+            .response(StatusCode::OK)
+            .json(json!({
+                "status": "not_found_in_tracks"
+            }))
+            .finish();
+        let app = TestAppBuilder::new()
+            .db_pool(db_pool.clone())
+            .core_client(core.into())
+            .build();
+        let small_infra = create_small_infra(&mut db_pool.get_ok()).await;
+
+        let request = app.post("/train_schedule/track_occupancy").json(
+            &json!({"train_schedule_ids": [train_schedule_valid.id],
+                "operational_point_id": "South_West_station",
+                "infra_id": small_infra.id
+            }),
+        );
+        let response: HashMap<String, Vec<TrackOccupancy>> =
+            app.fetch(request).assert_status(StatusCode::OK).json_into();
+        assert_eq!(response.len(), 0);
+    }
+
+    #[rstest]
+    #[case("T2", 50, Direction::StartToStop, vec![0, 100], vec![0, 1000], true, "T2")]
+    #[case("T2", 150, Direction::StartToStop, vec![0, 100], vec![0, 1000], false, "")]
+    #[case("T2", 50, Direction::StopToStart, vec![0, 100], vec![0, 1000], true, "T2")]
+    #[allow(clippy::too_many_arguments)]
+    fn test_interpolate_track_occupancy(
+        #[case] track_name: &str,
+        #[case] offset: u64,
+        #[case] direction: Direction,
+        #[case] positions: Vec<u64>,
+        #[case] times: Vec<u64>,
+        #[case] should_be_some: bool,
+        #[case] expected_track: &str,
```

Comment:
suggestion: These 2 can probably be merged into something like that
```suggestion
        #[case] expected_track: Option<&str>,
```